catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/ExportCommand.java

```diff
@@ -0,0 +1,569 @@
+            "-o"})
+    String output = Paths.get(System.getProperty("ddf.home"), name.get())
+            .toString();
+
+    @Option(name = "--delete", required = true, aliases = {"-d",
+            "delete"}, multiValued = false, description = "Whether or not to delete metacards after export")
+    boolean delete = false;
+
+    @Option(name = "--archived", required = false, aliases = {"-a",
+            "archived"}, multiValued = false, description = "Equivalent to --cql \"\\\"metacard-tags\\\" like 'deleted'\"")
+    boolean archived = false;
+
+    @Override
+    protected Object executeWithSubject() throws Exception {
+        Filter filter = getFilter();
+        transformer = getTransformers(DEFAULT_TRANSFORMER_ID).stream()
+                .findFirst()
+                .orElseThrow(() -> new CatalogCommandRuntimeException(
+                        "Could not find transformer " + DEFAULT_TRANSFORMER_ID));
+        revisionFilter = initRevisionFilter();
+
+        final File outputFile = initOutputFile(output);
+        if (outputFile.exists()) {
+            printErrorMessage(String.format("File [%s] already exists!", outputFile.getPath()));
+            return null;
+        }
+
+        final File parentDirectory = outputFile.getParentFile();
+        if (parentDirectory == null || !parentDirectory.isDirectory()) {
+            printErrorMessage(String.format("Directory [%s] must exist.", output));
+            console.println("If the directory does indeed exist, try putting the path in quotes.");
+            return null;
+        }
+
+        String filename = FilenameUtils.getName(outputFile.getPath());
+        if (StringUtils.isBlank(filename) || !filename.endsWith(".zip")) {
+            console.println("Filename must end with '.zip' and not be blank");
+            return null;
+        }
+
+        SecurityLogger.audit("Called catalog:dump command with path : {}\nCurrent user: {}",
+                output,
+                SubjectUtils.getName(SecurityUtils.getSubject()));
+
+        // TODO (RCZ) - Warn if breaking associations
+        ZipFile zipFile = new ZipFile(outputFile);
+
+        console.println("Starting metacard export...");
+        Instant start = Instant.now();
+        List<ExportItem> exportedItems = doMetacardExport(zipFile, filter);
+        console.println("Metacards exported in in: " + Duration.between(start, Instant.now()));
+        console.println("Number of metacards exported: " + exportedItems.size());
+        console.println();
+
+        console.println("Starting content export...");
+        start = Instant.now();
+        List<ExportItem> exportedContentItems = doContentExport(zipFile, exportedItems);
+        console.println("Content exported in: " + Duration.between(start, Instant.now()));
+        console.println("Number of content exported: " + exportedContentItems.size());
+
+        console.println();
+
+        if (delete) {
+            doDelete(exportedItems, exportedContentItems);
```

Comment:
until we absolutely need the speed, I'd rather not pre-optimize and muddy the code with concurrency.